pyls/_utils.py

```diff
@@ -9,21 +9,49 @@
 
 FIRST_CAP_RE = re.compile('(.)([A-Z][a-z]+)')
 ALL_CAP_RE = re.compile('([a-z0-9])([A-Z])')
+KWD_MARK = object()
 
 
-def debounce(interval_s):
+def debounce(interval_s, keys=None):
     """Debounce calls to this function until interval_s seconds have passed."""
     def wrapper(func):
+        arg_cache = {}
+        lock = threading.Lock()
+
+        @functools.wraps(func)
+        def cleanup(*args, **kwargs):
+            input_hash = _hash_input(keys, *args, **kwargs)
+            with lock:
+                del arg_cache[input_hash]
+            return func(*args, **kwargs)
+
         @functools.wraps(func)
```

Comment:
Remove the `wraps` decoration

Reply:
Actually, this is wrong. We should leave it on debounced and remove it from cleanup. The decorator basically copies the docstrings / other properties over. No point having it on cleanup. It's debounced that we return